### ISSUE TITLE
More docs regarding directives

### ISSUE THREAD

**Issue by joel-apollo** [planet42/Laika#85]
I'm trying to create some block directives to use in markdown. Call me crazy, but in a small scope, trying to create conditionals. 
The documentation has the following example:
```
@:for "documents.sections": { 
  <li>{{title.content}}</li>
} ~empty: {
  <p>no sections</p>
}
```
However with a block directive like this, this doesn't work:
```
val ifDirective: Blocks.Directive = Blocks.create("ifCondition") {
    (attribute("condition").optional ~ body(Default) ~ body("else").optional) {
      (condition, body, elseBody) =>
        OptionalDiv(condition.getOrElse(""), body, elseBody.getOrElse(Seq()))
    }
  }
```
The following markup does work (no brackets):
```
@:if condition=my_name==Joel : This is shown if you're Joel 
```
This doesn't parse properly
```
@:if condition=my_name==Joel : { 
    This is shown if you're Joel 
} ~else: { 
    You're not Joel!
}
```

More docs would be appreciated, as well as a way to debug this.

**Comment by jenshalm** [comment]
Many thanks for the feedback! This is indeed a very good spot, the docs are insufficient in that area. Directives seem to be a heavily underused feature, as you are the first to notice about 6 years after they've been introduced!

The aspect that is missing in the docs is that the boundaries for body elements of directives are different for span directives and block directives. For spans it is curly braces as documented. But for blocks, the body boundaries of the directive are marked by indentation:

```
@:if condition=my_name==Joel : 
  This is part of the directive body.
  This, too

This isn't
```

Now I realized I haven't thought about this design decision in a long time and I feel like it's counter-intuitive, even if documented properly. I think I made that decision because a closing brace can be placed anywhere, whereas a block cannot end in the middle of a line. But it could be made a condition that block directives end on a closing brace on an otherwise empty line, and then what you tried to do would actually work. 

In short, I'm thinking about changing the implementation instead of changing the docs. I'll do some analysis and tests shortly and then report back. If I change the implementation I will create a new ticket and reference it from here. If not I'll fix the docs instead.

**Comment by jenshalm** [comment]
I added a ticket for aligning the implementation with the documentation here: #87.

**Comment by jenshalm** [comment]
Master now contains a revised version of the documentation for directives which does no longer have any mismatch with the actual supported syntax.

But master is for version 0.12, which has an overhauled syntax for directives, removing some of the older inconsistencies. For 0.11 you can stick with the correction given in this ticket.
